src/vm/assembly.cpp

```diff
@@ -1983,6 +2010,51 @@ BOOL Assembly::IsInstrumentedHelper()
 }
 #endif // FEATURE_PREJIT
 
+
+#ifdef FEATURE_COMINTEROP
+
+ITypeLib* Assembly::GetTypeLib()
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_TRIGGERS;
+        FORBID_FAULT;
+    }
+    CONTRACTL_END
+
+    ITypeLib *pTlb = m_pITypeLib;
+    if (pTlb != nullptr && pTlb != (ITypeLib*)-1)
+        pTlb->AddRef();
+
+    return pTlb;
+} // ITypeLib* Assembly::GetTypeLib()
+
+void Assembly::SetTypeLib(ITypeLib *pNew)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_TRIGGERS;
+        FORBID_FAULT;
+    }
+    CONTRACTL_END
+
+    ITypeLib *pOld = InterlockedExchangeT(&m_pITypeLib, pNew);
```

Comment:
Should this rather be InterlockedCompare exchange that let's just the first one win? Looks like there is a potential for a race where the second one wins, we release the first one - but by the time we release it, somebody will be using it already.

Reply:
I noticed this wasn't using the compare, but wasn't sure it was intended to be this way or not. This code is basically a copy from Framework. But since you are asking, I am going to go with it should be the compare version.